2_process.R

```diff
@@ -30,5 +31,12 @@ p2_targets_list <- list(
       create_site_list(wqp_data_subset,p1_nwis_sites,p1_daily_data,p1_inst_data,
                        hucs=drb_huc8s,crs_out="NAD83",fileout = "2_process/out/DRB_SC_sitelist.csv")
     },
-    format = "file")
+    format = "file"),
+
+  tar_target(
+     p2_sites_w_segs,
+     {
+       sites_tbl <- read_csv(p2_site_list_csv)
+       get_site_flowlines(p1_reaches_sf, sites_tbl)
```

Comment:
Code for figures:
>plot(p1_reaches_sf[1], col = 'gray')
>plot(p1_reaches_sf[1][is.na(p1_reaches_sf$segidnat),], col = 'orange', add = T)
>plot(p2_sites_w_segs[is.na(p2_sites_w_segs$offset),], cex=0.2, pch = 16, col = 'red', add=T)
>plot(p2_sites_w_segs[which(is.na(p2_sites_w_segs$segidnat) & !is.na(p2_sites_w_segs$offset)),], cex=0.2, pch = 16, col = 'blue', add=T)

Reply:
I also converted the p2 variable to spatial points. I used sp methods because that's what I knew to do fast, but want to transition my knowledge to sf. Let me know if you want that code, too

Reply:
Thanks for catching these @jds485.

After some digging I finally realized what was happening. Those blue points are matching to the orange lines. But the orange lines don't have a `segidnat`, so when the join happens later in the function, there's not a value there, but there is for `subsegid`. If I change the code to match to `subsegid` instead of `segidnat` this is what we get (no blue points).

![image](https://user-images.githubusercontent.com/6943441/146273808-475b0f5c-e37e-4578-8680-cc73811b1935.png)

## `subsegid` vs. `segidnat`
For some reason there are 3 "segments" `(3, 8, 51)` that get split into two subsegs `(3_1, 3_2, 8_1, 8_2, 51_1, 51_2)`. For each of these, only the `*_2` subsegs get's matched with a `segidnat`.

```R
> segs %>% filter(subsegseg %in% c(3, 8, 51))
Simple feature collection with 6 features and 7 fields
Geometry type: LINESTRING
Dimension:     XY
Bounding box:  xmin: -75.08418 ymin: 40.8239 xmax: -74.50649 ymax: 42.26009
Geodetic CRS:  WGS 84
  subsegid subsegseg subseglen fromsegs toseg tosubseg segidnat                       geometry
1      3_1         3  14212.94        1    NA      3_2       NA LINESTRING (-74.59319 42.26...
2      3_2         3   5405.24        8     4      4_1     1437 LINESTRING (-74.62881 42.15...
3      8_1         8  17676.98     <NA>    NA      8_2       NA LINESTRING (-74.50649 42.03...
4      8_2         8   2050.96        2     3      3_2     1442 LINESTRING (-74.61166 42.14...
5     51_1        51    105.14       43    NA     51_2       NA LINESTRING (-75.08338 40.82...
6     51_2        51    460.51       50    52     52_1     1485 LINESTRING (-75.08399 40.82...
```


## proposal
The problem is that the meteorological data is matched to `segidnat`. So we can't just use `subsegid`.  I propose we match to `subsegid` then we also include `segidnat` in the resulting table and we can use either or both in the downstream processing. That way at least we have the matching if we find that we want it later on.

OR we could just match to segidnats (removing the segments with no `segidnat`)

For reference, it's only a very small percentage of data that does not match when using `segidnat` so I don't think it will be very consequential:

```
> blue_sites <- p2_sites_w_segs[which(is.na(p2_sites_w_segs$segidnat) & !is.na(p2_sites_w_segs$offset)),]
> sum(blue_sites$count_days_discrete)/sum(p2_sites_w_segs$count_days_discrete)
[1] 0.003990165
```

Reply:
Thanks for figuring this out! The 3 NA `segidnat` reaches must not be used in the model, right? If we assign points to their `_2` reach, would that match the logic of assigning to the outlet of reaches? Looks like it would based on the routing table columns. If that sounds good, it can be a post-processing step, rather than an edit made to these functions.

Reply:
> The 3 NA segidnat reaches must not be used in the model, right?

I would be surprised if these reaches aren't included in the model. I note these segments in issue #21 - subsegid 3_1 and 3_2 are plotted there for visual reference. These three reaches were split in the DRB processing pipeline and it just seems like the `segidnat` column didn't get populated for those split reaches. 

I agree with Jeff's proposal to match to `subsegid` since that is best suited to the goal of matching each point to the closest modeling reach. We should be able to fill in those `segidnat` NA values based on shared `subsegseg` ids for the purposes of linking the met data.

Reply:
The DRB temperature modeling reach attribute table also omits these 3 (456 total reaches in that table). [table](https://www.sciencebase.gov/catalog/item/5f6a289982ce38aaa2449135)